src/iexamine.cpp

```diff
@@ -1836,7 +1837,7 @@ void iexamine::kiln_full(player &, const tripoint &examp)
     }
 
     item result( "charcoal", calendar::turn.get_turn() );
-    result.charges = total_volume * char_type->ammo->def_charges / char_type->volume;
+    result.charges = total_volume / char_type->volume;
```

Comment:
OK, the current values are too high.

But there is still a problem here after those changes: volume per unit is `volume / stack_size`, but in the version here it is treated just as `volume`.